lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp

```diff
@@ -43,14 +43,51 @@
  *
  */
 
+#include <functional>
 #include "lazy_tensor_core/csrc/ts_backend/LazyShapeDtype.h"
 #include "aten/src/ATen/WrapDimUtils.h"
 #include "torch/csrc/api/include/torch/enum.h"
-
 namespace torch_lazy_tensors {
 namespace ir {
 namespace ops {
 
+std::vector<std::vector<int64_t>> compute_shape_std(const at::Tensor & self, bool unbiased){
+  return compute_shape_std(self, c10::nullopt, c10::nullopt, false);
+}
+std::vector<std::vector<int64_t>> compute_shape_std(const at::Tensor & self, at::IntArrayRef dim, bool unbiased, bool keepdim){
+  return compute_shape_std(self, dim, c10::nullopt, keepdim);
+}
+std::vector<std::vector<int64_t>> compute_shape_std(const at::Tensor & self, c10::optional<at::IntArrayRef> dim, c10::optional<int64_t> correction, bool keepdim){
+  if (dim.has_value()) {
+    auto sizes = self.sizes().vec();
+    if (keepdim) {
+      for (auto reduce_dim:  dim.value()) {
+        sizes[at::maybe_wrap_dim(reduce_dim, self)] = 1;
+      }
+      return {sizes};
+    } else {
+      std::vector<int64_t> reduce_dims(dim.value().begin(), dim.value().end());
+      at::maybe_wrap_dims(reduce_dims, self.sizes().size());
```

Comment:
Just `self` is fine?

Reply:
you mean 
at::maybe_wrap_dims(reduce_dims, self) ?

Reply:
Yea.

Reply:
oh. no, it doesn't work that way.  maybe_wrap_dims is unfortunately not defined in the same way as maybe_wrap_dim.  It's defined like this:

```
template <typename Container>
inline void maybe_wrap_dims(Container& dims, int64_t dim_post_expr) {
  return maybe_wrap_dims_n(dims.data(), dims.size(), dim_post_expr);
}
```

Reply:
Sad, then maybe `self.dim()`?

Reply:
ahh yes. better. thanks!